src/test/groovy/graphql/schema/GraphQLTypeReferenceTest.groovy

```diff
@@ -0,0 +1,37 @@
+package graphql.schema
+
+import graphql.Scalars
+import spock.lang.Specification
+
+class GraphQLTypeReferenceTest extends Specification {
+
+    def "the same reference can be used multiple times"() {
```

Comment:
maybe more descriptive: doesn't throw exception. Or make the assertion more clear and check really that the type rerference can be replaced.